pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -578,7 +579,11 @@ private void addNewRow(GenericRow row)
           // Update inverted index
           RealtimeInvertedIndexReader invertedIndex = indexContainer._invertedIndex;
           if (invertedIndex != null) {
-            invertedIndex.add(dictId, docId);
+            try {
+              invertedIndex.add(dictId, docId);
```

Comment:
For `RealtimeInvertedIndexReader.add()`, we should also handle the case of `_bitmaps.size() < dictId` which is possible when the inverted index is not updated for the previous record. We can fill empty bitmaps for the missing `dictId`s.

Reply:
I don't think this is possible any more - all exceptions are caught

Reply:
This is now impossible because nothing can fail in between the dictionary id being updated and `RealtimeInvertedIndexReader.add()`, so the inverted index will always be updated if the dictId is incremented, unless it's `RealtimeInvertedIndexReader.add()` which fails, but it can't fail if it never skips values.